src/Command/Util/ApigeeEdgeManagementCliService.php

```diff
@@ -0,0 +1,330 @@
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\TransferException;
+use Symfony\Component\Console\Style\StyleInterface;
+
+/**
+ * Defines an interface for Edge connection classes.
+ */
+class ApigeeEdgeManagementCliService implements ApigeeEdgeManagementCliServiceInterface {
+
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  protected $httpClient;
+
+  /**
+   * Constructs a ApigeeEdgeManagementCliService object.
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The HTTP client.
+   */
+  public function __construct(ClientInterface $http_client) {
+    $this->httpClient = $http_client;
+  }
+
+  /**
+   * Create role in Apigee Edge for Drupal to use for Edge connection.
+   *
+   * @param \Symfony\Component\Console\Style\StyleInterface $io
+   *   The IO interface of the CLI tool calling the method.
+   * @param callable $t
+   *   The translation function akin to t().
+   * @param string $org
+   *   The organization to connect to.
+   * @param string $email
+   *   The email of an Edge user with org admin role to make Edge API calls.
+   * @param string $password
+   *   The password of an Edge user with org admin role to make Edge API calls.
+   * @param null|string $base_url
+   *   The base url of the Edge API.
+   * @param null|string $role_name
+   *   The role name to add the permissions to.
+   */
+  public function createEdgeRoleForDrupal(StyleInterface $io,
+                                          callable $t,
+                                          string $org,
+                                          string $email,
+                                          string $password,
+                                          ?string $base_url,
+                                          ?string $role_name) {
+
+    if ($base_url === NULL || $base_url === '') {
+      // Set default base URL if var is null or empty string.
+      $base_url = self::DEFAULT_BASE_URL;
+    }
+    else {
+      // Validate it is a valid URL.
+      if (!UrlHelper::isValid($base_url, TRUE)) {
+        $io->error($t('Base URL is not valid.'));
+        return;
+      }
+
+    }
+
+    if ($role_name === NULL || $role_name === '') {
+      // Set default if null or empty string.
+      $role_name = self::DEFAULT_ROLE_NAME;
+    }
+
+    if (!$this->isValidEdgeCredentials($io, $t, $org, $email, $password, $base_url)) {
+      return;
+    }
+
+    $does_role_exist = $this->doesRoleExist($org, $email, $password, $base_url, $role_name);
+
+    // If role does not exist, create it.
+    if (!$does_role_exist) {
+      $io->text($t('Role :role does not exist. Creating role.', [':role' => $role_name]));
+
+      $url = "{$base_url}/o/{$org}/userroles";
+      try {
+        $this->httpClient->post($url, [
+          'body' => json_encode([
+            'role' => [$role_name],
+          ]),
+          'auth' => [$email, $password],
```

Comment:
This won't support oauth-enabled orgs I think. Is there a way to use the php-sdk instead here, which should handle that for you?

Reply:
new issue #274 will handle this.